### DISCUSSION TITLE
Select all checkboxes?

### DISCUSSION THREAD

**Discussion by vladstudio** [ariakit/ariakit#3860] [Questions] [answered]
Hey!
In this example:
https://ariakit.org/examples/checkbox-group

how do I programmatically select all (none) checkboxes? I must be missing something obvious. Thanks!

**Comment by vladstudio** [comment]
Forgot to add: I am using `useCheckboxStore` like this:

```
const [value, setValue] = useState([])
const store = useCheckboxStore({ value, setValue: setValue })

<CheckboxProvider store={store} ....
```

**Comment by diegohaz** [comment, marked as answer]
If you're managing state with `React.useState`, you can handle it just like native checkboxes:

Select all:

```jsx
setValue(["apple", "orange", "mango"]); // or store all the values elsewhere
```

Select none:
```jsx
setValue([]);
```

**Reply by vladstudio** [reply]
I just found I was passing a wrong value to setValue, that's why it was not working :-) Thanks! Works perfectly.

(And thanks for Ariakit in general!)
